php-frontend/src/test/java/org/sonar/php/cache/CacheTest.java

```diff
@@ -156,4 +181,9 @@ private static InputFile inputFile(String content) {
       .setContents(content)
       .build();
   }
+
+  private static DefaultInputFile notExistingFile() {
```

Comment:
```suggestion
  private static DefaultInputFile fileWithoutContent() {
```
You want to test the use case of a file without content. So you provide a file. Maybe `notExistingFile` is not suitable.

Reply:
Not exactly. The file with empty content produces a valid hash - I added a test case for that. 
Here I test the situation that a file doesn't exist (wrong configuration, deletion of a file during the scan?). The execution will probably fail earlier before the `Cache` code is executed. 
Anyway, I would avoid unnecessary cache write with a `null` hash to be on the safe side.

Reply:
Okay got it, but you use the method inside the `shouldNotWriteToCacheWhenNoContent` test case. So maybe rename the test case.